packages/crayons-core/src/components/icon/icon.tsx

```diff
@@ -11,41 +17,168 @@ import icons from './icon-assets/icons.json';
+   */
+  @Prop() width: number;
+  /**
+   * Height of the icon, specified in number of  pixels.
+   */
+  @Prop() height: number;
   /**
    * Color in which the icon is displayed, specified as a standard CSS color or as a HEX code.
    */
   @Prop() color = '';
+  /**
+   * Enable Intersection Observer. Default is false.
+   */
+  @Prop() lazy = false;
+  /**
+   * Name of External Library to be used
+   */
+  @Prop() library = 'crayons';
+
+  @Element() el: HTMLElement;
+
+  @State() private visible = false;
 
-  @State() svgHTML = '';
+  @State() private intersectionObserver: IntersectionObserver;
+
+  @State() private svgHTML = '';
+
+  @State() private eval_xOb = false;
+
+  @State() private parser = new DOMParser();
 
   @Watch('name')
-  private setSVGState(iconName: string) {
-    this.svgHTML = icons[iconName];
+  @Watch('src')
+  @Watch('library')
+  private async setSVGState(): Promise<void> {
+    const { name, visible, library, lazy, dataSvg, src } = this;
+
+    if (!lazy) this.eval_xOb = true;
+    else this.eval_xOb = visible;
+
+    if (!Build.isBrowser || !this.eval_xOb) {
+      return;
+    }
+    if (!name && dataSvg) {
+      this.svgHTML = dataSvg;
+    } else if (name) {
+      try {
+        let path = src;
+        if (library !== 'crayons' && !src) {
+          path = this.getUrl();
+        }
+        const lib = src ? 'cdn' : library;
+        if (library === 'crayons' || path !== undefined) {
+          const div = document.createElement('div');
+          div.innerHTML = await fetchIcon(name, lib, path);
+
+          const svgEle = div.firstElementChild;
+          const svg =
+            svgEle && svgEle.tagName.toLowerCase() === 'svg'
+              ? svgEle.outerHTML
+              : '';
+          const doc = this.parser.parseFromString(svg, 'text/html');
+          const svgEl = doc.body.querySelector('svg');
+
+          const iconlibrary = getIconLibrary(this.library);
+          if (iconlibrary && iconlibrary.mutator) {
+            iconlibrary.mutator(svgEl, name);
+          }
+
+          this.svgHTML = svgEl.outerHTML;
+        } else {
+          throw 'URL Not Valid.';
+        }
+      } catch (ex) {
+        console.info(
+          `Cannot load ${name}|${library} from CDN. Please check the url & proide a live path.`,
+          ex
+        );
+        this.loadFallbackImage();
+      }
+    } else {
+      console.error(
+        "Please provide valid props 'name' or 'icon'.Check the usage docs."
+      );
+      return;
+    }
   }
 
-  componentWillLoad() {
-    this.setSVGState(this.name);
+  private getUrl(): string {
+    const library = getIconLibrary(this.library);
+    if (this.name && library) {
+      return library.resolver();
+    } else {
+      return this.src;
+    }
   }
 
-  render() {
-    const style = {
-      '--icon-color': this.color,
-    };
+  async loadFallbackImage() {
+    this.svgHTML = await fetchIcon('image', 'crayons', undefined);
```

Comment:
You can avoid passing the 3rd argument instead of passing it as undefined.

Reply:
Fixed